File: test/fixture/expo-module/android/src/main/java/expo/vscode.example/ExampleModuleModule.kt
package expo.vscode.example

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExampleModuleModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExampleModule")

    AsyncFunction("helloAsync") { options: Map<String, String> ->
      println("Hello 👋")
    }

    ViewManager {
      View { context -> 
        ExampleModuleView(context) 
      }

      Prop("name") { view: ExampleModuleView, prop: String ->
        println(prop)
      }
    }
  }
}

File: test/fixture/schema-expo-module/ios/ExampleModuleModule.swift
import ExpoModulesCore

public class ExampleModuleModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExampleModule")

    AsyncFunction("helloAsync") { (options: [String: String]) in
      print("Hello 👋")
    }

    ViewManager {
      View {
        ExampleModuleView()
      }

      Prop("name") { (view: ExampleModuleView, prop: String) in
        print(prop)
      }
    }
  }
}
